sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedReadFromBoundedSource.java

```diff
@@ -208,10 +208,10 @@ public void encode(Checkpoint<T> value, OutputStream outStream, Context context)
       private boolean done;
 
       public Reader(
-          @Nullable List<TimestampedValue<T>> residualElements,
+          List<TimestampedValue<T>> residualElements,
           @Nullable BoundedSource<T> residualSource,
           PipelineOptions options) {
-        this.residualElements = residualElements;
+        this.residualElements = Preconditions.checkNotNull(residualElements, "residualElements");
```

Comment:
Test is failing - you have to pass the empty list in, or Objects.firstNotNull it